### PR TITLE
Drop password and confirmationcode

### PR DESCRIPTION
**Pull request by mstriemer**
These columns will no longer be needed after we drop unconfirmed users in https://github.com/mozilla/addons-server/issues/3746.

### CODE REVIEW

File: src/olympia/users/models.py [hunk 1]
```diff
@@ -124,6 +121,7 @@ class UserProfile(OnChangeMixin, ModelBase, AbstractBaseUser):
     notifycompat = models.BooleanField(default=True)
     notifyevents = models.BooleanField(default=True)
     occupation = models.CharField(max_length=255, default='', blank=True)
+    password = None  # We don't store passwords.
```
Comment (EnTeQuAk) [inline src/olympia/users/models.py]: Maybe add a link to the fxa tracking github-issue for documentation reasons?

File: src/olympia/migrations/907-drop-password-and-confirmationcode.sql [hunk 2]
```diff
@@ -0,0 +1,3 @@
+ALTER TABLE users
+    DROP COLUMN `password`,
+    DROP COLUMN `confirmationcode`;
```
Comment (EnTeQuAk) [inline src/olympia/migrations/907-drop-password-and-confirmationcode.sql]: I think this is likely to produce some kind of load or hanging on prod, let's see how it goes on dev and stage and keep @jasonthomas in the loop that there's a bigger migration coming :)

Comment (bqbn) [inline src/olympia/migrations/907-drop-password-and-confirmationcode.sql]: 👍  for cleaning up the unused stuff btw.

**Comment by mstriemer** [comment]
It actually looks like dropping the password field will be a bit more work, I'll submit a new PR for that.

**Comment by mstriemer** [comment]
I went ahead and updated our user model to not include the `password` column by copying over django's `AbstractBaseUser` and removing the password related code. That change is included in the https://github.com/mozilla/addons-server/pull/3866/files/6320702ac080848097fcf85df5dad38ba1410fbf..5a33927053333a84da11d2f73784a796678d3b7diff.

File: src/olympia/migrations/907-drop-password-and-confirmationcode.sql (hunk 2 above)
Comment (EnTeQuAk) [inline src/olympia/migrations/907-drop-password-and-confirmationcode.sql]: In theory just a few seconds, I don't have too much data locally to verify unfortunately. I'd take a closer look on `-dev` and `-stage` to make sure it stays "a few seconds".

**Comment by mstriemer** [comment]
It looks like master is failing in the same way now so I guess something else is breaking the build. I'd imagine this is safe to merge pending a review of the updated user model, it works for me locally and the tests pass.

File: Makefile [hunk 3]
```diff
@@ -223,6 +224,12 @@ ifneq ($(IN_DOCKER),)
 endif
 	docker exec -t -i ${DOCKER_NAME} ./manage.py shell
 
+dbshell:
+ifneq ($(IN_DOCKER),)
+	$(warning Command is designed to be run in the host)
+endif
+	docker exec -t -i ${DOCKER_NAME} ./manage.py dbshell
+
```
Comment (EnTeQuAk) [inline Makefile]: Not really part of this pull request but small enough so okay. Best to split those things up next time.

File: src/olympia/migrations/907-drop-password-and-confirmationcode.sql (hunk 2 above)
Comment (diox) [inline src/olympia/migrations/907-drop-password-and-confirmationcode.sql]: Please add a note to the push etherpad when you merge this, so that we can anticipate any issues

File: Makefile (hunk 3 above)
Comment (mstriemer) [inline Makefile]: Yeah, I was planning on keeping it as a separate commit. I won't squash it with the rest of it.

**Comment by mstriemer** [comment]
Well for some reason this is breaking the setup-docker.sh script. It looks like the add-ons are being created without being flagged as supporting Firefox. I'm not sure what that has to do with the user model not having a password but maybe I'm missing something.

**Comment by EnTeQuAk** [comment]
Not sure where this is coming from but this is what I see in the circleci log

```
Traceback (most recent call last):
  File "manage.py", line 11, in <module>
    execute_from_command_line(sys.argv)
  File "/usr/lib/python2.7/site-packages/django/core/management/__init__.py", line 354, in execute_from_command_line
    utility.execute()
  File "/usr/lib/python2.7/site-packages/django/core/management/__init__.py", line 346, in execute
    self.fetch_command(subcommand).run_from_argv(self.argv)
  File "/usr/lib/python2.7/site-packages/django/core/management/base.py", line 394, in run_from_argv
    self.execute(*args, **cmd_options)
  File "/usr/lib/python2.7/site-packages/django/core/management/base.py", line 445, in execute
    output = self.handle(*args, **options)
  File "/usr/lib/python2.7/site-packages/django/contrib/staticfiles/management/commands/collectstatic.py", line 168, in handle
    collected = self.collect()
  File "/usr/lib/python2.7/site-packages/django/contrib/staticfiles/management/commands/collectstatic.py", line 107, in collect
    handler(path, prefixed_path, storage)
  File "/usr/lib/python2.7/site-packages/django/contrib/staticfiles/management/commands/collectstatic.py", line 315, in copy_file
    self.storage.save(prefixed_path, source_file)
  File "/usr/lib/python2.7/site-packages/django/core/files/storage.py", line 63, in save
    name = self._save(name, content)
  File "/usr/lib/python2.7/site-packages/django/core/files/storage.py", line 222, in _save
    os.makedirs(directory)
  File "/usr/lib64/python2.7/os.py", line 150, in makedirs
    makedirs(head, mode)
  File "/usr/lib64/python2.7/os.py", line 157, in makedirs
    mkdir(name, mode)
OSError: [Errno 13] Permission denied: '/code/site-static/css'
```

I'll try restarting it but I remember seeing this before and it's unrelated to your PR imho.

did you already rebase with master?

**Comment by mstriemer** [comment]
Yeah, I rebased with master. I spent way too much time trying to debug this yesterday. If you checkout this branch and run `docker-compose run web ./scripts/setup-docker.sh` it will wipe the db and the add-ons it creates won't appear on the homepage.

The build that passed works for me locally as well. I tried figuring out why it isn't creating the add-ons properly but there doesn't appear to be any errors (other than the errors that occur on master too and seem to be image related).

**Comment by mstriemer** [comment]
Looks like the error is back to what I'm seeing locally.

**Comment by mstriemer** [comment]
@EnTeQuAk I couldn't figure out how to fix the setup-docker.sh script when the user doesn't have a password and thought it might have other consequences that aren't being tested. I left the password column in and updated the migration to empty out the password column instead of dropping it.

**Comment by mstriemer** [comment]
Fixed in https://github.com/mozilla/addons-server/commit/2842d6135a675c8176c717065ede16e83653b80d.
